Publish to Maven Central and make it work with Spark 3.0.1

arielshaqed opened the pull request:
Pushing to keep working; might not _entirely_ work... :-/

arielshaqed committed:
Downgrade Hadoop, Google Commons versions to use with Spark-3.0.1

johnnyaug commented on project/plugins.sbt:
```diff
@@ -1 +1,2 @@
 addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.15.0")
+addSbtPlugin("com.github.sbt" % "sbt-pgp" % "2.1.2")
```
Do we want this?

arielshaqed replied:
[The fine manual says we do](https://github.com/sbt/sbt-pgp#setup)!  I imagine the reasoning is something like this...
> _I know I can sign and publish with sbt-pgp version 2.1.2.  I might not be able to sign and publish with some other version of sbt-pgp.  So let's drag it in._
Given that we do the same with sbt-assembly, which also says to include it, I think this makes sense.

johnnyaug commented on build.sbt:
```diff
@@ -24,11 +24,11 @@ core / libraryDependencies ++= Seq("org.rocksdb" % "rocksdbjni" % "6.6.4",
-  "org.apache.hadoop" % "hadoop-aws" % "2.10.1",
-  "org.apache.hadoop" % "hadoop-common" % "2.10.1",
+  "org.apache.hadoop" % "hadoop-aws" % "2.7.2",
+  "org.apache.hadoop" % "hadoop-common" % "2.7.2",
```
2.7.2 is very old. I'm working with 2.7.7 and it works (for now)

arielshaqed committed:
Separate assembly of parts

With this, the core jar from `sbt assembly` can be used in `spark-shell` like this:
```sh
spark-shell --master local --jars ./core/target/scala-2.12/core-assembly-0.1.0-SNAPSHOT.jar
--conf spark.hadoop.lakefs.api.url=http://some.lakefs.server.dev/api/v1  --conf
spark.hadoop.lakefs.api.access_key=AKIA...lakefs --conf
spark.hadoop.lakefs.api.secret_key=shhh...lakefs --conf
spark.hadoop.fs.s3a.access.key=AKIA...S3 --conf spark.hadoop-fs.s3a.secret.key=shhh...S3

scala> import io.treeverse.clients._
scala> val df = LakeFSContext.newDF(spark, "repo", "ref")
scala> df.take(10)
```

arielshaqed commented:
Thanks!
